src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,491 @@
+        ),
+      minuteStep,
+    );
+  }, [minuteStep, selectedTime]);
+
+  const seconds = React.useMemo(() => {
+    const time = selectedTime ?? new Date();
+    return generateDataList(
+      60,
+      (i: number) =>
+        new Date(
+          time.getFullYear(),
+          time.getMonth(),
+          time.getDate(),
+          time.getHours(),
+          time.getMinutes(),
+          i,
+        ),
+      secondStep,
+    );
+  }, [secondStep, selectedTime]);
+
+  return (
+    <div className={cx('iui-time-picker', className)} {...rest}>
+      <TimePickerColumn
+        data={hours}
+        isSameFocused={(val) =>
+          isSameHour(val, focusedTime, use12Hours ? meridiem : undefined)
+        }
+        isSameSelected={(val) =>
+          isSameHour(val, selectedTime, use12Hours ? meridiem : undefined)
+        }
+        onFocusChange={onHourFocus}
+        onSelectChange={onHourClick}
+        setFocus={setFocusHour}
+        valueRenderer={hourRenderer}
+      />
+      {precision != 'hours' && (
+        <TimePickerColumn
+          data={minutes}
+          isSameFocused={(val) => isSameMinute(val, focusedTime)}
+          isSameSelected={(val) => isSameMinute(val, selectedTime)}
+          onFocusChange={(date) => setFocusedTime(date)}
+          onSelectChange={(date) => updateCurrentTime(date)}
+          valueRenderer={minuteRenderer}
+        />
+      )}
+      {precision == 'seconds' && (
+        <TimePickerColumn
+          data={seconds}
+          isSameFocused={(val) => isSameSecond(val, focusedTime)}
+          isSameSelected={(val) => isSameSecond(val, selectedTime)}
+          onFocusChange={(date) => setFocusedTime(date)}
+          onSelectChange={(date) => updateCurrentTime(date)}
+          valueRenderer={secondRenderer}
+        />
+      )}
+      {use12Hours && (
+        <TimePickerColumn<MeridiemType>
+          data={['AM', 'PM']}
+          isSameFocused={(val) => isSameMeridiem(val, focusedTime)}
+          isSameSelected={(val) => isSameMeridiem(val, selectedTime)}
+          onFocusChange={(date) => onMeridiemFocus(date)}
+          onSelectChange={(value) => onMeridiemClick(value)}
+          valueRenderer={meridiemRenderer}
+          className='iui-period'
+        />
+      )}
+    </div>
+  );
+};
+
+type TimePickerColumnProps<T = Date> = {
+  /**
+   * Data to render in column.
+   */
+  data: T[];
+  /**
+   * Compare function for focus.
+   */
+  isSameFocused: (value: T) => boolean;
+  /**
+   * Compare function for select.
+   */
+  isSameSelected: (value: T) => boolean;
+  /**
+   * Callback when focus is changed.
+   */
+  onFocusChange: (value: T) => void;
+  /**
+   * Callback when date is changed.
+   */
+  onSelectChange: (value: T) => void;
+  /**
+   * Set initial focus.
+   */
+  setFocus?: boolean;
+  /**
+   * What value to display in every cell.
+   */
+  valueRenderer: (value: T) => JSX.Element;
+  /**
+   * Class to apply on root.
+   */
+  className?: string;
```

Comment:
Could use classNameProps? Or even StylingProps

Reply:
Wrote it that way as inner component and its mostly for the iui-period reason.. but I can take classNameProps I guess..

Reply:
using classNameProps